engine/card.py

```diff
@@ -22,9 +25,33 @@ def __str__(self):
     def __repr__(self):
         return str(self)
 
-    def make_public(self, information_type):
+    def matches(self, valid_colors=None, valid_numbers=None):
+        return self.does_color_match(valid_colors) and self.does_number_match(valid_numbers)
+
+    def does_color_match(self, valid_colors=None):
+        if isinstance(valid_colors, Card):
+            valid_colors = valid_colors.color
+        return not valid_colors or self.color.matches(valid_colors)
+
+    def does_number_match(self, valid_numbers=None):
```

Comment:
I'm not strongly convinced we need to genericize / wild-enable numbers.  This might be interesting for thought experiment stuff, but I can't think of a reason this is super interesting when an AI is unlikely to take this into account with heuristics. Additionally, if we WERE doing this, we should make a HanabiNumber class.  Rather save this until we have an explicit use case in mind.